### PR TITLE
Better intervals between feelers

### PR DESCRIPTION
**Pull request by naumenkogs**
~_Draft pending on #19958_~

1. If an attempt to connect to a feeler wasn't actually made (an entry from AddrMan we considered was invalid, or we made too many tries, or it didn't have the right flag) — wait less until we try the next feeler.
2. Make those intervals mockable
3. Remove unnecessary extra random noise between feelers

(1) would make transitioning from "new" to "tried" better in the presence of those events. An adversary here fills our "new" table with garbage to prevent us from trying honest peers from "new". This probably won't help much against a very powerful adversary but will help against a moderate adversary. It also may help in natural non-malicious cases.

Moving from "new" to "tried" table is important for p2p security.

### CODE REVIEW

**Comment by sdaftuar** [comment]
Concept ACK improvements along these lines.  However, I'm not sure about the more complicated interval logic around the feeler connections -- it seems to me like that logic is already a bit hard to follow, so I wonder if a better approach might be to just check for invalidity in the `while (!interruptNet)` loop, and be willing to iterate in that loop many more times if `fFeeler` is set?

**Comment by naumenkogs** [comment]
@sdaftuar I agree with the high-level idea. Even though I don't think I'm making it significantly harder to follow, simplifying is a good idea.

But let me check, are you suggesting just to carefully use `continue` instead of `break` inside the loop?

**Comment by naumenkogs** [comment]
Closing this:
1. Mockable time is already done.
2. Smarter intervals probably better be done after a refactoring (gonna do that in another PR first)
3. Dropping extra noise probably ain't worth it on its own
